src/main/java/com/ericsson/ei/rules/RulesHandler.java

```diff
@@ -115,21 +147,49 @@ public RulesObject getRulesForEvent(String event) {
         return null;
     }
 
-    private String getContent(InputStream inputStream){
-        try {
-
+    /**
+     * Reads the rules file content from a URI.
+     *
+     * @return rules file content
+     * @throws IOException
+     * @throws URISyntaxException
+     */
+    private String readRulesFileFromURI() throws IOException, URISyntaxException {
+        return IOUtils.toString(new URI(rulesFilePath), "UTF-8");
+    }
 
-            ByteArrayOutputStream result = new ByteArrayOutputStream();
-            byte[] buffer = new byte[1024];
-            int length;
-            while ((length = inputStream.read(buffer)) != -1) {
-                result.write(buffer, 0, length);
-            }
-            return result.toString("UTF-8");}
-        catch (Exception e) {
-            LOGGER.error(e.getMessage(), e);
+    /**
+     * Reads the rules file content from a full or relative path.
+     *
+     * @return rules file content
+     * @throws IOException
+     * @throws Exception
+     */
+    private String readRulesFileFromPath() throws IOException {
+        String rulesJsonFileContent = null;
+        InputStream inputStream = this.getClass().getResourceAsStream(rulesFilePath);
+        if (inputStream == null) {
+            rulesJsonFileContent = FileUtils.readFileToString(new File(rulesFilePath), Charset.defaultCharset());
+        } else {
+            rulesJsonFileContent = IOUtils.toString(inputStream, "UTF-8");
```

Comment:
inputStream should be closed before returning from this method.

Reply:
I tried closing it and put the logic in a try-finally block but I got a "IllegalState Failed to load ApplicationContext" error when I did that... maybe something is going on in the toString method in IOUtils.

Reply:
@evasiba-ericsson showed me that you can use InputStream as a try resource since the close method implements AutoCloseable and that seems to work. So this is done.